packages/backend/src/auth/auth.service.ts

```diff
@@ -61,79 +62,103 @@ export const signup = async (req: express.Request, res: express.Response) => {
     const existingUser = await User.query().where('email', email).first()
 
     if (existingUser) {
-      return res.status(409).json({ error: 'User already exists' })
+      return res
+        .status(409)
+        .json({ message: 'User already exists', success: false })
     }
     await User.query().insert({ email, password })
 
-    return res.status(201).json({ message: 'Success' })
+    return res.status(201).json({ message: 'Success', success: true })
   } catch (error) {
     log.error(error)
-    return res.status(500).json({ error: 'Unable to create user' })
+    return res
+      .status(500)
+      .json({ message: 'Unable to create user', success: false })
   }
 }
 
-export const login = async (req: express.Request, res: express.Response) => {
-  const { email, password } = await zParse(loginSchema, req)
+export const login = async (
+  req: Request,
+  res: Response<BaseResponse>,
+  next: NextFunction
+) => {
+  try {
+    const { email, password } = await zParse(loginSchema, req)
 
-  const user = await User.query()
-    .findOne({ email })
-    .select('id', 'email', 'password')
-    .throwIfNotFound(new UnauthorisedException('Invalid credentials'))
-    .withGraphFetched('refreshTokens')
+    const user = await User.query()
+      .findOne({ email })
+      .select('id', 'email', 'password')
+      .withGraphFetched('refreshTokens')
 
-  const isValid = await user.verifyPassword(password)
-  if (!isValid) {
-    throw new UnauthorisedException('Invalid credentials')
-  }
+    if (!user) {
+      throw new UnauthorisedException('Invalid credentials')
+    }
 
-  let refreshToken = user.refreshTokens?.[0]
+    const isValid = await user.verifyPassword(password)
+    if (!isValid) {
+      throw new UnauthorisedException('Invalid credentials')
+    }
 
-  const { refreshToken: generatedToken, expiresIn: refreshTokenExpiresIn } =
-    generateRefreshToken(user.id)
+    let refreshToken = user.refreshTokens?.[0]
 
-  if (refreshToken) {
-    refreshToken.token = generatedToken
-    refreshToken.expiresAt = new Date(Date.now() + refreshTokenExpiresIn * 1000)
+    const { refreshToken: generatedToken, expiresIn: refreshTokenExpiresIn } =
+      generateRefreshToken(user.id)
 
-    await refreshToken.$query().patch(refreshToken)
-  } else {
-    refreshToken = new RefreshToken()
-    refreshToken.token = generatedToken
-    refreshToken.expiresAt = new Date(Date.now() + refreshTokenExpiresIn * 1000)
+    if (refreshToken) {
+      refreshToken.token = generatedToken
+      refreshToken.expiresAt = new Date(
+        Date.now() + refreshTokenExpiresIn * 1000
+      )
 
-    await user.$relatedQuery('refreshTokens').insert(refreshToken)
-  }
+      await refreshToken.$query().patch(refreshToken)
+    } else {
+      refreshToken = new RefreshToken()
+      refreshToken.token = generatedToken
+      refreshToken.expiresAt = new Date(
```

Comment:
us a method that transforms `expiresIn` to `expiresAt`
or maybe one that gets the refresh token and applies the new values (token + expiresAt)

Reply:
Good point. 
I used the RefreshToken constructor for this purpose and modified the code in auth service to use it accordingly.